packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -700,6 +700,14 @@ export default function(
     instance.refs = emptyObject;
     instance.context = getMaskedContext(workInProgress, unmaskedContext);
 
+    warning(
+      instance.state !== instance.props,
+      '%s: It is not recommended to assign props directly to state' +
+        "because updates to props won't be reflected in state." +
+        'In most cases, it is better to use props directly.',
+      getComponentName(workInProgress) || 'A component',
+    );
```

Comment:
This needs to be within an `if (__DEV__)` check. Suggest using the one right below (line 711).

Reply:
Fixed!